src/scripting/lua_map.cc

```diff
@@ -4782,18 +4873,337 @@ int LuaConstructionSite::get_building(lua_State* L) {
 	return 1;
 }
 
+/* RST
+   .. attribute:: has_builder
+
+      (RW) Whether this constructionsite has a builder. Changing this setting causes the worker to be instantly deleted or to be created from thin air.
+*/
+int LuaConstructionSite::get_has_builder(lua_State* L) {
+	lua_pushboolean(L, get(L, get_egbase(L))->builder_.is_set());
+	return 1;
+}
+int LuaConstructionSite::set_has_builder(lua_State* L) {
+	Widelands::EditorGameBase& egbase = get_egbase(L);
+	Widelands::PartiallyFinishedBuilding& cs = *get(L, egbase);
+	const bool arg = luaL_checkboolean(L, -1);
+	if (cs.builder_.is_set()) {
+		if (!arg) {
+			cs.builder_.get(egbase)->remove(egbase);
+			cs.builder_ = nullptr;
+			cs.set_seeing(false);
+			cs.request_builder(egbase);
+		}
+	} else if (arg) {
+		assert(cs.builder_request_);
+		delete cs.builder_request_;
+		cs.builder_request_ = nullptr;
+		Widelands::Worker& w = egbase.tribes()
+		                          .get_worker_descr(cs.owner().tribe().builder())
+		                          ->create(egbase, cs.get_owner(), &cs, cs.get_position());
+		cs.builder_ = &w;
+		if (upcast(Widelands::Game, g, &egbase)) {
+			w.start_task_buildingwork(*g);
+		}
+		cs.set_seeing(true);
+	}
+	return 0;
+}
+
+/* RST
+   .. attribute:: setting_launch_expedition
+
+      (RW) Only valid for ports under construction. Whether an expedition will be launched immediately upon completion.
+*/
+int LuaConstructionSite::get_setting_launch_expedition(lua_State* L) {
+	upcast(Widelands::WarehouseSettings, ws, get(L, get_egbase(L))->get_settings());
+	assert(ws);
+	lua_pushboolean(L, ws->launch_expedition);
+	return 1;
+}
+int LuaConstructionSite::set_setting_launch_expedition(lua_State* L) {
+	upcast(Widelands::WarehouseSettings, ws, get(L, get_egbase(L))->get_settings());
+	assert(ws);
+	ws->launch_expedition = luaL_checkboolean(L, -1);
+	return 0;
+}
+
+/* RST
+   .. attribute:: setting_stopped
+
+      (RW) Only valid for productionsites and trainingsites under construction. Whether this building will be initially stopped after completion.
+*/
+int LuaConstructionSite::get_setting_stopped(lua_State* L) {
+	upcast(Widelands::ProductionsiteSettings, ws, get(L, get_egbase(L))->get_settings());
+	assert(ws);
```

Comment:
You cannot assert these, because you don't control the Lua code that people will write. You'll have to check and report_error.